app/services/register_and_partner_api/sync_users.rb

```diff
@@ -20,27 +20,39 @@ def self.perform(all: false)
       last_sync = SyncUsersTimer.last_sync
       base_query = all ? {} : { filter: { updated_since: last_sync } }
 
-      # is_last_page = false
-      page_number = 0
-      # until is_last_page || page_number > 10_000
-      page_number += 1
-      paginated_query = base_query.merge(page: { page: page_number, per_page: 100 })
-
-      response = RegisterAndPartnerApi::User.where(paginated_query)
-      sync_users(response)
-
-      # is_last_page = true if response.count.zero?
-      # end
+      perform_pagination(base_query)
 
       SyncUsersTimer.set_last_sync(new_sync_time)
     rescue StandardError => e
       Rails.logger.warn("Failed to sync users: #{e}")
     end
 
+    def self.perform_pagination(base_query)
+      is_last_page = false
+      page_number = 0
+      all_user_ids = []
+      until is_last_page
+        page_number += 1
+        paginated_query = base_query.merge(page: { page: page_number, per_page: 10 })
```

Comment:
Why 10 per page?

Reply:
Cause it made testing easier and I forgot to take it out.